ui/webui/src/components/Error.jsx

```diff
@@ -16,55 +16,135 @@
     Modal,
+    ModalVariant,
+    TextArea,
     TextContent,
     TextVariants,
     Text,
 } from "@patternfly/react-core";
+import { ExternalLinkAltIcon } from "@patternfly/react-icons";
 
 import { exitGui } from "../helpers/exit.js";
 
 const _ = cockpit.gettext;
 
-export const CriticalError = ({ exception, isBootIso }) => {
+export const bugzillaPrefiledReportURL = (productQueryData) => {
+    const baseURL = "https://bugzilla.redhat.com";
+    const queryData = {
+        ...productQueryData,
+        component: "anaconda",
+    };
+
+    const reportURL = new URL(baseURL);
+    reportURL.pathname = "enter_bug.cgi";
+    Object.keys(queryData).map(query => reportURL.searchParams.append(query, queryData[query]));
+    return reportURL.href;
+};
+
+const addExceptionDataToReportURL = (url, exception) => {
+    const newUrl = new URL(url);
+
+    const context = exception.contextData?.context ? exception.contextData?.context + " " : "";
+
+    newUrl.searchParams.append(
+        "short_desc",
+        "WebUI: " + context + exception.name + ": " + exception.message
+    );
+    newUrl.searchParams.append(
+        "comment",
+        "Installer WebUI Critical Error:\n" + context + exception.name + ": " + exception.message + "\n\n" + _("Please attach the file /tmp/webui.log to the issue.")
+    );
+    return newUrl.href;
+};
+
+export const CriticalError = ({ exception, isBootIso, reportLinkURL }) => {
+    const reportURL = addExceptionDataToReportURL(reportLinkURL, exception);
+    const [logContent, setLogContent] = useState("");
+    const [preparingReport, setPreparingReport] = useState(false);
+
+    useEffect(() => {
+        cockpit.spawn(["journalctl", "-a"])
+                .then(content => setLogContent(content));
+    }, []);
+
+    const openBZIssue = (reportURL) => {
+        setPreparingReport(true);
+        cockpit
+                .file("/tmp/webui.log")
+                .replace(logContent)
+                .always(() => setPreparingReport(false))
+                .then(() => window.open(reportURL, "_blank", "noopener,noreferer"));
+    };
+
+    const context = exception.contextData?.context;
+
     return (
         <Modal
-          description={_("The installer cannot continue due to a critical error.")}
+          description={context
+              ? cockpit.format(_("The installer cannot continue due to a critical error: $0"), _(context))
+              : _("The installer cannot continue due to a critical error.")}
           id="critical-error-modal"
           isOpen
           position="top"
           showClose={false}
           title={_("Critical error")}
           titleIconVariant="danger"
-          variant="small"
+          variant={ModalVariant.large}
           footer={
               <>
+                  {reportLinkURL &&
+                  <Button
+                    variant="primary"
+                    isLoading={preparingReport}
+                    isDisabled={preparingReport}
```

Comment:
You should disable this button instead of adding sleep in the test. So isDisabled={!content || preparingReport}

Reply:
I'll open a followup PR.